### DISCUSSION TITLE
Crash with arm64/aarch64 Linux configuration

### DISCUSSION THREAD

**Discussion by artun42** [ppy/osu#23791] [Q&A]
### Type

Crash to desktop

### Bug description

This is a duplicate of #7166. (The hardware has changed slightly since then.)

I'm running Linux on a MacBook with M1 SoC. By default, the Linux kernel operates with a 16K page size on the M1. Knowing this, I switched to a 4K pages Linux kernel and placed the `librealm-wrappers.so` file, which I downloaded from the upstream artifact, in `~/osu/osu.Desktop/bin/dotnet/Debug/net6.0/runtimes/linux-arm64/native/.` According to [this PR](https://github.com/realm/realm-dotnet/pull/3267), it appears to be possible to use realm-dotnet with an arm64/aarch64 Linux host now. However, in the real world, osu!lazer still cannot run and crashes with database corruption. I don't expect it to work under a 16K page kernel configuration (due to realm, but osu!lazer clearly can handle this), but a 4K pages configuration should be able to run lazer. 

### Version

2023.607.0 (git)

### Logs

[logs.zip](https://github.com/ppy/osu/files/11672104/logs.zip)

(There have been multiple attempts to run this, resulting in several corrupt databases.)

**Comment by natsukagami** [comment]
Update: I managed to compile and run the current lazer version (2023.1114.1) on Macbook Air M1 with NixOS (`nixos-apple-silicon` patches). 
It requires the latest Mesa version with OpenGL 3.3 implementation (I'm compiling from the `asahi-20231121` tag of the fork).

It's actually quite painless now. Steps are as follows:
- Grab a clone of https://github.com/veldrid/veldrid-spirv and compile it (`./build-native.sh -release linux-x64` would do, you'd still get aarch64 libs). Add the output dir to the `LD_LIBRARY_PATH` env.
- Get libbass libraries (bass, bass_fx and bassmix) and add them to library path.

That should be enough to compile and run actually... I think `librealm-wrappers` are now packaged automatically (`LD_DEBUG=libs` shows the library loaded from `/osu.Desktop/bin/Debug/net6.0/runtimes/linux-arm64/native/librealm-wrappers.so`).
